packages/core/status/core-status-server-internal/src/log_plugins_status.test.ts

```diff
@@ -0,0 +1,585 @@
+            },
+          ],
+
+          degraded: [
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyDegradedPlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'alreadyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+          ],
+
+          unavailable: [],
+        },
+        total: 2,
+        updated: 2,
+        updates: {
+          available: [
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.available,
+              pluginName: 'newlyAvailablePluginDependantPlugin2',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+          ],
+
+          critical: [],
+          degraded: [
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'newlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPlugin',
+              summary: 'summary',
+            },
+            {
+              level: ServiceStatusLevels.degraded,
+              pluginName: 'anotherNewlyDegradedPluginDependantPlugin1',
+              summary: 'summary',
+              isInferredStatus: true,
+            },
+          ],
+
+          unavailable: [],
+        },
+      })
+    ).toMatchInlineSnapshot(`
```

Comment:
In this test we have a very explanatory example of how the "diff status" events are processed, and what things are logged.